### DISCUSSION TITLE
Here with a ForeignKey constraint then [May 26]

### DISCUSSION THREAD

**Discussion by dstala** [nocodb/nocodb#5809] [Discord Threads]
> 🙋‍♂️ **64gfh7393gfj49**  
## Here with a ForeignKey constraint then Parent field in Child table should be populated with ID  
![image0](https://cdn.discordapp.com/attachments/661905455894888493/1011604351313248266/Screenshot_2022-08-23_at_12.52.20_pm.png)  
![image1](https://cdn.discordapp.com/attachments/661905455894888493/1011604351749459968/Screenshot_2022-08-23_at_12.52.25_pm.png)

**Comment by o1lab** [comment]
> 💁 **o1lab**  
#### Im struggling to understand what is missing.   
####    
#### Seems like you have created the relation correctly.   
####    
#### Parent A is associated with one child and Parent B in similar way    
####    
#### What are you expecting ? what is missing ?  
#  
> 🙋‍♂️ **64gfh7393gfj49**  
#### So instead of manually adding each child to the parent using the GUI I expect to be able to infer the relationship from a field of ForeignKeys which in this case is called "Parent"  
#  
> 🙋‍♂️ **64gfh7393gfj49**  
#### When I import the Parent table form a CSV I already have the ForeignKey information but NocoDB expects to to manually rekey it all in  
#  
> 🙋‍♂️ **64gfh7393gfj49**  
#### No option for related field to existing table  
![image0](https://cdn.discordapp.com/attachments/1011604353150373928/1011624448417341541/Screenshot_2022-08-23_at_2.12.17_pm.png)  
#  
> 🙋‍♂️ **64gfh7393gfj49**  
#### And then after importing I can not select the LinkToAnotherRecord option either for the field  
#  
> 💁 **o1lab**  
#### ah, you mean thsi for uploading bulk and creating relations with it   
#### I understand that is important, but this is slightly tricky than it feels   
#### hence the upload is in alpha   
#### the reason being : this looked-up or related-field can come from any nested level  
#  
> 🙋‍♂️ **64gfh7393gfj49**  
#### okay thanks at least I understand there are limitations now not just I don't understand 🙂  
#  
> 🙋‍♂️ **64gfh7393gfj49**  
#### I expected to be able to just type in or paste into the ForeignKey field without the manual GUI selection process  
#  
> 💁 **o1lab**  
#### thank you for the understanding 🙂  
#  
> 🙋‍♂️ **Eric**  
#### i think it's important to not that defining links from child to parent doesn't behave in the system as a valid foreign key, since the child key can only be use by one parent (exclusive).   this is not a behaviour expected from a foreign key  
#  
> 🙋‍♂️ **Eric**  
#### the previously implemented pattern of setting up a foreign key value from the parent is the *correct* modeling paradigm.  
#  
> 💁 **o1lab**  
#### LinkToAnotherRecord  
#  
> 💁 **o1lab**  
#### Has to be created in parent table  
#  
> 💁 **o1lab**  
#### With had many  
#  
> 💁 **o1lab**  
#### Has  
#  
> 🙋‍♂️ **Eric**  
#### i'll try again to confirm  
#  
> 🙋‍♂️ **Eric**  
#### ok, i build another example db and tested.  
#  
> 🙋‍♂️ **Eric**  
#### confirmed that the creating the key from the parent with 'has many' creates an exclusive  one to many relationship (e.g. the linked key value can only be related to  one and only one parent record).  
#  
> 🙋‍♂️ **Eric**  
#### structurally, the pattern that is missing is:  from a table (parent) no option for many-to-one.  this is the textbook foreign key value to a domain of valid keys.  
#  
> 🙋‍♂️ **Eric**  
#### e.g.   i will have many "parent" records with an attribute pointing to a primary key in another table representing a domain of valid values.  
#  
> 💁 **o1lab**  
#### for this you go to another table to create hasMany 😄 ?  
#  
> 💁 **o1lab**  
#### will not that solve it  
#  
> 🙋‍♂️ **Eric**  
#### no...   that then makes it so that the keys in the "another table" can only be references by a single entity in the related  
#  
> 🙋‍♂️ **Eric**  
####   
[firefox_QoQtNbWjjs.mp4](https://cdn.discordapp.com/attachments/1011604353150373928/1012103362101137488/firefox_QoQtNbWjjs.mp4)  
#  
> 💁 **o1lab**  
#### LifeCycle_State    
#### LifeCycle2   
####    
#### There are two columns - are those columns coming from same table ? (looks like)  
#  
> 💁 **o1lab**  
#### can u help describe this is in an github issue  
#  
> 💁 **o1lab**  
#### a fullscreen video will make it really clear  
#  
> 🙋‍♂️ **Eric**  
#### sure.  will do.  
#  
> 💁 **o1lab**  
#### thank you  
#  
> 🙋‍♂️ **Eric**  
#### question.  
#  
> 🙋‍♂️ **Eric**  
#### if a sqllite db implements the relationship correctly, will the user interface respect it?  
#  
> 🙋‍♂️ **Eric**  
#### i'll provide the equivalent DDL for the sqllite as well.  
#  
> 💁 **o1lab**  
#### should work  
#  
> 💁 **o1lab**  
#### we infer from existing databases as well  
#  

 [Autogenerated from discord](https://discord.com/channels/661905455894888490/1011604353150373928)  
- - - -  
#### Join [NocoDB's](http://www.nocodb.com/)  community  
[Discord](https://discord.gg/5RgZmkW) • [Twitter](https://twitter.com/nocodb) • [Reddit](https://www.reddit.com/r/NocoDB/) • [Docs](https://docs.nocodb.com/)
